File: Cap 08/8.3 - Matrizes de Duas Dimensoes/8.3.1 e 8.3.2 juntos/mat2d.py
'''
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capítulo ..: 8 - Topico 8.3
 *
 * Programa ..: ENTRADA E SAIDA EM MATRIZ 2D
'''

NOTAS = [[0.0] * 4 for _ in range(8)]

print("MATRIZ 2D - ENTRADA E SAIDA\n")
for I in range(8):
  print(f"Entre as notas do aluno {I + 1}:")
  for J in range(4):
    print(f"Nota ==> {J + 1}: ", end="")
    NOTAS[I][J] = float(input())
  print()

print("\nRELATORIO DE NOTAS\n")
print("Aluno Nota1 Nota2 Nota3 Nota4")
print("----- ----- ----- ----- -----")
for I in range(8):
  print(f"{I + 1:5d}", end="")
  for J in range(4):
      print(f"{NOTAS[I][J]:6.1f}", end="")
  print()

File: Cap 08/8.3 - Matrizes de Duas Dimensoes/8.3.1 e 8.3.2 juntos/mat2d.scala
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capítulo ..: 8 - Topico 8.3
 * 
 * Programa ..: ENTRADA E SAIDA EM MATRIZ 2D
 * 
 */

object Main extends App {
  val NOTAS: Array[Array[Double]] = Array.ofDim[Double](8, 4)

  println("MATRIZ 2D - ENTRADA E SAIDA\n")
  for (I <- 0 to 7 by 1) {
    println(s"Entre as notas do aluno ${I + 1}:")
    for (J <- 0 to 3 by 1) {
      print(s"Nota ==> ${J + 1}: ")
      NOTAS(I)(J) = scala.io.StdIn.readDouble()
    }
    println()
  }

  println("\nRELATORIO DE NOTAS\n")
  println("Aluno Nota1 Nota2 Nota3 Nota4")
  println("----- ----- ----- ----- -----")
  for (I <- 0 to 7 by 1) {
    print(f"${I + 1}%5d")
    for (J <- 0 to 3 by 1) {
      print(f" ${NOTAS(I)(J)}%5.1f")
    }
    println()
  }
  
}

